source/grid/grid_generator.cc

```diff
@@ -7090,6 +7090,273 @@ namespace GridGenerator
+
+            // large tets parallel y axis
+            add_cell(dim, {0, 2, 8, 12}, current_material_id);
+            add_cell(dim, {4, 6, 13, 8}, current_material_id);
+            add_cell(dim, {5, 13, 7, 9}, current_material_id);
+            add_cell(dim, {1, 9, 3, 12}, current_material_id);
+
+            // large tets parallel z axis
+            add_cell(dim, {0, 8, 4, 10}, current_material_id);
+            add_cell(dim, {1, 5, 9, 10}, current_material_id);
+            add_cell(dim, {3, 7, 11, 9}, current_material_id);
+            add_cell(dim, {2, 6, 8, 11}, current_material_id);
+
+            // large tets in middle octaeder
+            add_cell(dim, {12, 13, 10, 9}, current_material_id);
+            add_cell(dim, {12, 13, 9, 11}, current_material_id);
+            add_cell(dim, {12, 13, 11, 8}, current_material_id);
+            add_cell(dim, {12, 13, 8, 10}, current_material_id);
+
+            // small tets in corners
+            add_cell(dim, {13, 8, 10, 4}, current_material_id);
+            add_cell(dim, {13, 10, 9, 5}, current_material_id);
+            add_cell(dim, {13, 9, 11, 7}, current_material_id);
+            add_cell(dim, {13, 11, 8, 6}, current_material_id);
+
+            add_cell(dim, {10, 12, 9, 1}, current_material_id);
+            add_cell(dim, {9, 12, 11, 3}, current_material_id);
+            add_cell(dim, {11, 12, 8, 2}, current_material_id);
+            add_cell(dim, {8, 12, 10, 0}, current_material_id);
+          }
+        else
+          {
+            Assert(false, ExcNotImplemented());
+          }
+
+        // Set up sub-cell data.
+        for (const auto f : cell.face_indices())
+          {
+            const auto current_boundary_id = cell.face(f)->boundary_id();
+            if (current_boundary_id == numbers::internal_face_boundary_id)
+              continue;
+            if (dim == 2) // 2D boundary_lines
+              {
+                if (cell.face(f)->boundary_id() !=
+                    numbers::internal_face_boundary_id) // face at boundary
+                  {
+                    if (f == 0)
+                      {
+                        add_cell(1, {0, 4}, current_boundary_id);
+                        add_cell(1, {4, 2}, current_boundary_id);
+                      }
+                    else if (f == 1)
+                      {
+                        add_cell(1, {1, 5}, current_boundary_id);
+                        add_cell(1, {5, 3}, current_boundary_id);
+                      }
+                    else if (f == 2)
+                      {
+                        add_cell(1, {0, 6}, current_boundary_id);
+                        add_cell(1, {6, 1}, current_boundary_id);
+                      }
+                    else if (f == 3)
+                      {
+                        add_cell(1, {2, 7}, current_boundary_id);
+                        add_cell(1, {7, 3}, current_boundary_id);
+                      }
+                  }
```

Comment:
There's an
```
  else
    Assert (false, ExcInternalError());
```
case missing here.

This whole sequence of if-else-if statements could have been written as a switch-case statement as well, and that's the style we generally use. (Same for the 3d case below)

Reply:
Alternatively, you could have just put the various indices into one big table of the form
```
  const unsigned int face_to_new_vertex_indices[4][2][2] = ...;
```
and then just call
```
  add_cell (1, face_to_new_vertex_indices[f][0], current_boundary_id);
  add_cell (1, face_to_new_vertex_indices[f][1], current_boundary_id);
```